runtime/jvmti/jvmtiThread.c

```diff
@@ -1331,8 +1340,8 @@ jvmtiSuspendResumeCallBack(J9VMThread *vmThread, J9MM_IterateObjectDescriptor *o
 					}
 				} else {
 					/* NULL carrier thread imply the virtual threadis unmounted. */
-					if (0 == J9OBJECT_U32_LOAD(vmThread, vthread, vm->isSuspendedByJVMTIOffset)) {
-						J9OBJECT_U32_STORE(vmThread, vthread, vm->isSuspendedByJVMTIOffset, 1);
+					if (0 == J9OBJECT_U32_LOAD(vmThread, vthread, vm->isSuspendedInternalOffset)) {
+						J9OBJECT_U32_STORE(vmThread, vthread, vm->isSuspendedInternalOffset, 1);
 					}
 				}
```

Comment:
The halt flag should only be set if the vthread is mounted. `isSuspendedInternal` should always be set.
```
	if ((NULL != targetThread) && (vthread == targetThread->threadObject)) {
		...
	}
	J9OBJECT_U32_STORE(vmThread, vthread, vm->isSuspendedInternalOffset, 1);
```

Reply:
As guided, added code to set the halt flag (J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND) only if the thread
is mounted, i.e., threadObject == targetThread->threadObject in J9VMThread's publicFlags